crypto/cpuid.c

```diff
@@ -125,71 +125,22 @@ void OPENSSL_cpuid_setup(void)
-                } else {
-                    OPENSSL_ia32cap_P[2] = (unsigned int)vecx;
-                    OPENSSL_ia32cap_P[3] = (unsigned int)(vecx >> 32);
-                }
-            }
-        }
-
-        if (env && (((env = ossl_strchr(env, ':')) != NULL))) {
-            env++;
-            if ((env != NULL) && (env[0] != '\0') && (env[0] != ':')) {
-                IA32CAP vecx;
-
-                off = (env[0] == '~') ? 1 : 0;
-                vecx = ossl_strtouint64(env + off);
-                if (off) {
-                    OPENSSL_ia32cap_P[4] &= ~(unsigned int)vecx;
-                    OPENSSL_ia32cap_P[5] &= ~(unsigned int)(vecx >> 32);
-                } else {
-                    OPENSSL_ia32cap_P[4] = (unsigned int)vecx;
-                    OPENSSL_ia32cap_P[5] = (unsigned int)(vecx >> 32);
-                }
-            }
-        }
-
-        if (env && (((env = ossl_strchr(env, ':')) != NULL))) {
-            env++;
-            if ((env != NULL) && (env[0] != '\0') && (env[0] != ':')) {
-                IA32CAP vecx;
-
-                off = (env[0] == '~') ? 1 : 0;
-                vecx = ossl_strtouint64(env + off);
-                if (off) {
-                    OPENSSL_ia32cap_P[6] &= ~(unsigned int)vecx;
-                    OPENSSL_ia32cap_P[7] &= ~(unsigned int)(vecx >> 32);
-                } else {
-                    OPENSSL_ia32cap_P[6] = (unsigned int)vecx;
-                    OPENSSL_ia32cap_P[7] = (unsigned int)(vecx >> 32);
-                }
-            }
-        }
-
-        if (env && (((env = ossl_strchr(env, ':')) != NULL))) {
-            env++;
-            if ((env != NULL) && (env[0] != '\0') && (env[0] != ':')) {
-                IA32CAP vecx;
-
-                off = (env[0] == '~') ? 1 : 0;
-                vecx = ossl_strtouint64(env + off);
-                if (off) {
-                    OPENSSL_ia32cap_P[8] &= ~(unsigned int)vecx;
-                    OPENSSL_ia32cap_P[9] &= ~(unsigned int)(vecx >> 32);
-                } else {
-                    OPENSSL_ia32cap_P[8] = (unsigned int)vecx;
-                    OPENSSL_ia32cap_P[9] = (unsigned int)(vecx >> 32);
+        int index = 2;
+        for (; index < OPENSSL_IA32CAP_P_MAX_INDEXES; index += 2) {
+            if (env && (((env = ossl_strchr(env, ':')) != NULL))) {
+                env++;
+                if ((env != NULL) && (env[0] != '\0') && (env[0] != ':')) {
+                    IA32CAP vecx;
+
+                    off = (env[0] == '~') ? 1 : 0;
+                    vecx = ossl_strtouint64(env + off);
+                    if (off) {
+                        OPENSSL_ia32cap_P[index] &= ~(unsigned int)vecx;
+                        OPENSSL_ia32cap_P[index + 1] &= ~(unsigned int)(vecx >> 32);
+                    } else {
+                        OPENSSL_ia32cap_P[index] = (unsigned int)vecx;
+                        OPENSSL_ia32cap_P[index + 1] = (unsigned int)(vecx >> 32);
+                    }
```

Comment:
good idea to create a loop across the indexes to address comment about code duplication